src/tracers.jl

```diff
@@ -224,16 +224,16 @@ end
 
 Convenience constructor for [`ConnectivityTracer`](@ref), [`GradientTracer`](@ref) and [`HessianTracer`](@ref) from input indices.
 """
-function create_tracer(::Type{Dual{P,T}}, primal::Number, index::Integer) where {P,T}
+function create_tracer(::Type{Dual{P,T}}, primal::Real, index::Integer) where {P,T}
     return Dual(primal, create_tracer(T, primal, index))
 end
 
-function create_tracer(::Type{GradientTracer{G}}, ::Number, index::Integer) where {G}
+function create_tracer(::Type{GradientTracer{G}}, ::Real, index::Integer) where {G}
     return GradientTracer{G}(sparse_vector(G, index))
 end
-function create_tracer(::Type{ConnectivityTracer{C}}, ::Number, index::Integer) where {C}
+function create_tracer(::Type{ConnectivityTracer{C}}, ::Real, index::Integer) where {C}
     return ConnectivityTracer{C}(sparse_vector(C, index))
 end
-function create_tracer(::Type{HessianTracer{G,H}}, ::Number, index::Integer) where {G,H}
+function create_tracer(::Type{HessianTracer{G,H}}, ::Real, index::Integer) where {G,H}
     return HessianTracer{G,H}(sparse_vector(G, index), empty(H))
 end
```

Comment:
Needs support for returning a `Complex{T}` when called with `Complex` number.

Reply:
#102